sky/provision/gcp/instance_utils.py

```diff
@@ -615,23 +615,42 @@ def create_or_update_firewall_rule(
     @classmethod
     def set_labels(cls, project_id: str, availability_zone: str, node_id: str,
                    labels: dict) -> None:
-        node = cls.load_resource().instances().get(
-            project=project_id,
-            instance=node_id,
-            zone=availability_zone,
-        ).execute(num_retries=GCP_CREATE_MAX_RETRIES)
-        body = {
-            'labels': dict(node['labels'], **labels),
-            'labelFingerprint': node['labelFingerprint'],
-        }
-        operation = (cls.load_resource().instances().setLabels(
-            project=project_id,
-            zone=availability_zone,
-            instance=node_id,
-            body=body,
-        ).execute(num_retries=GCP_CREATE_MAX_RETRIES))
-
-        cls.wait_for_operation(operation, project_id, zone=availability_zone)
+        retry_cnt = 0
+        backoff = common_utils.Backoff(initial_backoff=constants.POLL_INTERVAL)
+        while True:
```

Comment:
Is it possible to re-use `_retry_on_http_exception` decorator (defined in the same file) instead of implementing retry logic again here?

Reply:
Good catch! Updated it to use `_retry_on_http_exception` instead.